core/controllers/ingest.py

```diff
@@ -51,7 +56,8 @@ def ingest(body: dict, excel_file: FileStorage) -> Response:
         # when no reporting round, source workbook is already in a transformed state
         workbook = original_workbook  # TODO: Deprecate ability to ingest this workbook with no round.
 
-    validate(workbook, original_workbook, reporting_round)
+    if validation_failures := validate(workbook, original_workbook, reporting_round):
+        return process_validation_failures(validation_failures)
```

Comment:
I prefer this to raising an Exception - at least in this instance. But validation failures can be instantiated during PTV and transformation so we need to think about how we handle that as well

Reply:
I'm finding it very complex to unite these three paths into seperate sources of validation failures that can all benifit from a Messeger class.
- schema validation errors, and pre-transformation errors are fine
- TF round 4 and errors raised during transformation are much more complicated because they are constructed so differently